client/search-ui/src/input/SearchBox.tsx

```diff
@@ -68,6 +96,35 @@ export const SearchBox: React.FunctionComponent<React.PropsWithChildren<SearchBo
         [onEditorCreatedCallback]
     )
 
+    const onSearchHistorySelect = useCallback(
+        (search: RecentSearch) => {
+            const searchContext = getGlobalSearchContextFilter(search.query)
+            const query =
+                searchContext && searchContext.spec === selectedSearchContextSpec
+                    ? omitFilter(search.query, searchContext?.filter)
+                    : search.query
+            onChange({ query, hint: EditorHint.Focus })
+        },
+        [onChange, selectedSearchContextSpec]
+    )
+
+    const historyWithoutSelectedContext = useMemo(() => {
+        if (!recentSearches || !selectedSearchContextSpec) {
+            return undefined
+        }
+
+        return recentSearches.map(search => {
+            const searchContext = getGlobalSearchContextFilter(search.query)
+            if (searchContext && searchContext.spec === selectedSearchContextSpec) {
+                return {
+                    ...search,
+                    query: omitFilter(search.query, searchContext?.filter),
+                }
+            }
+            return search
+        })
+    }, [recentSearches, selectedSearchContextSpec])
```

Comment:
This was confusing to me. I suggest renaming to clarify that this is a modification of the existing `recentSearches` and adding comments to explain what this is doing.

```suggestion
    // Modify the recentSearches list to remove the context filter if it matches the
    // currently selected context, as it will already be present when submitting the search.
    const recentSearchesWithoutSelectedContext = useMemo(() => {
        if (!recentSearches || !selectedSearchContextSpec) {
            return undefined
        }

        return recentSearches.map(search => {
            const searchContext = getGlobalSearchContextFilter(search.query)
            if (searchContext && searchContext.spec === selectedSearchContextSpec) {
                return {
                    ...search,
                    query: omitFilter(search.query, searchContext?.filter),
                }
            }
            return search
        })
    }, [recentSearches, selectedSearchContextSpec])
```